python/az/aro/azext_aro/custom.py

```diff
@@ -273,9 +273,12 @@ def get_route_tables_from_subnets(cli_ctx, subnets):
     for sn in subnets:
         sid = parse_resource_id(sn)
 
-        subnet = network_client.subnets.get(resource_group_name=sid['resource_group'],
-                                            virtual_network_name=sid['name'],
-                                            subnet_name=sid['resource_name'])
+        if not all(key in sid.keys() for key in ('resource_group', 'name', 'resource_name')):
+            logger.warning("Failed to validate subnet %s", sn)
```

Comment:
I see the coments have been addressed only partially. I believe the error still should be returned, as it happened. 
I also further log should be provided to offer additional guide for customer.

So I would like to see Bens comment here.
1. Log warning with additional info about retries and support tickets, also log the error
2. Raise an error ValidationError, the error happened and it damages the deployment

Reply:
Added as suggested